modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -218,10 +218,17 @@ CV__DNN_INLINE_NS_BEGIN
          *  @param[out] outputs vector of already allocated output blobs
          *
          * If this method is called after network has allocated all memory for input and output blobs
-         * and before inferencing.
+         * and before layer fusion and inferencing.
          */
         CV_WRAP virtual void finalize(InputArrayOfArrays inputs, OutputArrayOfArrays outputs);
 
+        /** @brief After layer fusion, we need carry on some nassary process for layer.
+         *
+         * This method is called after network has allocated all memory for input and output blobs
+         * and layer fusion, but before inferencing.
+         */
+        CV_WRAP virtual void afterFuse();
```

Comment:
Let's not introduce this general method to Layer API. We are just optimizing convolution for now, let's do the changes locally for now

Reply:
Thanks, regarding depth wise, is there any bug in the code?